### ISSUE TITLE
REST service to list Rasa responses

### ISSUE THREAD

**Issue by turnerrainer** [buerokratt/Training-Module#90]
AS AN Architect
I WANT to have a REST service to fetch a list of all Rasa responses
SO THAT there wouldn't be duplicate services for it

### Acceptance Criteria

> Temporary list of endpoints - https://github.com/buerokratt/Buerokratt-onboarding/issues/8

**GET endpoint**

- [x] Use GET endpoint `/rasa/responses` to list all responses with response name and value
- [x] Make requests to OpenSearch matching Rasa forms with index matching [Rasa domain.yml](https://github.com/buerokratt/Training-Module/blob/main/mock1/domain.yml#L55-L113)

#### Example of input and output

`responses` of `domain.yml` as a raw input

> PS! Request to OpenSearch

```yaml
responses:
  utter_andmekaitse_küsimused:
  - text: "kõik on kaitstud"

  utter_fallback_kontroll:
  - text: "Jäi arusaamatuks, kas soovisite, et suunan edasi?"

  utter_mille_kohta_küsimus_käis:
  - text: "Kas küsimus käis selle teema kohta: {intent}? (Jah/Ei)"

  utter_not_confident:
  - text: "Ma ei saanud päris täpselt aru."
  
  utter_ei_pakkunud_õigesti:
  - text: "Ma pean natuke juurde õppima, et oskaksin sellele vastata."

  utter_ask_direct_to_customer_support_form_affirm_deny:
  - text: "Kas suunan teid klienditeenindajale? (Jah/Ei)"
```

Output

```json
{
  "responses": [{
    "name": "utter_andmekaitse_küsimused",
    "text": "kõik on kaitstud"
  },
  {
    "name": "utter_fallback_kontroll",
    "text": "Jäi arusaamatuks, kas soovisite, et suunan edasi?"
  },
  {
    "name": "utter_mille_kohta_küsimus_käis",
    "text": "Kas küsimus käis selle teema kohta: {intent}? (Jah/Ei)"
  },
  {
    "name": "utter_not_confident",
    "text": "Ma ei saanud päris täpselt aru."
  },
  {
    "name": "utter_ei_pakkunud_õigesti",
    "text": "Ma pean natuke juurde õppima, et oskaksin sellele vastata."
  },
  {
    "name": "utter_ask_direct_to_customer_support_form_affirm_deny",
    "text": "Kas suunan teid klienditeenindajale? (Jah/Ei)"
  }]
}
```

**POST endpoint**

- [ ] POST service `/rasa/responses` with a POST parameter `input` to fetch the full content of all matching responses within the scope of `response_name` and `response_text`

Where in

```yaml
  utter_ask_direct_to_customer_support_form_affirm_deny:
  - text: "Kas suunan teid klienditeenindajale? (Jah/Ei)"
```

`utter_ask_direct_to_customer_support_form_affirm_deny` corresponds to `response_name` and `Kas suunan teid klienditeenindajale? (Jah/Ei)` to `response_text`.

#### Example output for a request

See example `responses` of `domain.yml` as a raw input from top

> PS! Request to OpenSearch to search for `on` gives the following output:

```json
{
  "responses": [{
    "name": "utter_andmekaitse_küsimused",
    "text": "kõik on kaitstud"
  },
  {
    "name": "utter_fallback_kontroll",
    "text": "Jäi arusaamatuks, kas soovisite, et suunan edasi?"
  },
  {
    "name": "utter_not_confident",
    "text": "Ma ei saanud päris täpselt aru."
  }]
}
```

**Comment by RaulAltmae** [comment]
Is the Get '/rasa/responses' endpoint suppose to get data from opensearch or domain.yml?

**Comment by turnerrainer** [comment]
> Is the Get '/rasa/responses' endpoint suppose to get data from opensearch or domain.yml?

@RaulAltmae the AC currently states

> - [ ] Make requests to OpenSearch matching Rasa forms with index matching [Rasa domain.yml](https://github.com/buerokratt/Training-Module/blob/main/mock1/domain.yml#L55-L113)

plus

> **Example of input and output**

having a comment

> PS! Request to OpenSearch

Would you like to change that?

**Comment by RaulAltmae** [comment]
No, request to opensearch is good. There was confusion, because in endpoint task it states that 

> list all Rasa responses from domain.yml in responses

![image](https://user-images.githubusercontent.com/120466882/215984638-23ef9651-d37c-4387-aaa3-f309a5817f10.png)

Thats why i asked, just to be sure that we understand it the same way

**Comment by RaulAltmae** [comment]
This is blocked by https://github.com/buerokratt/Ruuter/discussions/132. If the problem with Ruuter is fixed then it should work as intended.

**Comment by turnerrainer** [comment]
> This is blocked by [buerokratt/Ruuter#132](https://github.com/buerokratt/Ruuter/discussions/132). If the problem with Ruuter is fixed then it should work as intended.

https://github.com/buerokratt/Ruuter/discussions/132#discussioncomment-4958365

**Comment by turnerrainer** [comment]
Blocked due to Node errors. @KasparMinn to fix, @varmoh deploys.

**Comment by turnerrainer** [comment]
Blocked by #289

**Comment by KasparMinn** [comment]
PR: https://github.com/buerokratt/Training-Module/pull/316

This PR changes the functionality of the responses page slightly to fetch correct information from OpenSearch as well as correctly establish the three different searches for Rules.

- Added a new DSL and Handlebars script to process the responses-list with all conditions and rules.
- Somewhat improved the front-end updating speed on Responses page to refresh on mutation success.
- Changed the symbol requirement for adding and updating intents from 6 to 3.

Screenshots of the requests under PR link.
